packages/playwright-test/types/test.d.ts

```diff
@@ -16,14 +16,18 @@
  */
 
 import type { APIRequestContext, Browser, BrowserContext, BrowserContextOptions, Page, LaunchOptions, ViewportSize, Geolocation, HTTPCredentials, Locator, APIResponse, PageScreenshotOptions } from 'playwright-core';
+import type { TestCase } from '@playwright/test/reporter';
+
 export * from 'playwright-core';
 
+export type junitClassNameTemplate = (test: TestCase) => string;
+
 export type ReporterDescription =
   ['dot'] |
   ['line'] |
   ['list'] |
   ['github'] |
-  ['junit'] | ['junit', { outputFile?: string, stripANSIControlSequences?: boolean }] |
+  ['junit'] | ['junit', { outputFile?: string, stripANSIControlSequences?: boolean, classNameTemplate?: junitClassNameTemplate }] |
```

Comment:
I'm not sure we want to allow functions in the configuration, we don't do it in other places today, maybe we can have a declarative config similar to [snapshotPathTemplate](https://playwright.dev/docs/api/class-testproject#test-project-snapshot-path-template). Let me discuss this in the team meeting tomorrow.